core/vulkan/vk_api_timing_layer/cc/layer_helpers.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "core/cc/target.h"
+#include "core/cc/log.h"
+extern "C" {
+__attribute__((constructor)) void _layer_dummy_func__();
+}
+#if (TARGET_OS == GAPID_OS_WINDOWS) || (TARGET_OS == GAPID_OS_OSX)
+class dummy_struct {};
+#else
+#include <dlfcn.h>
+#include <cstdio>
+#include <cstdint>
+class dummy_struct {
+    public:
+    dummy_struct();
+};
+
+dummy_struct::dummy_struct() {
+    GAPID_ERROR("Loading dummy struct");
+    Dl_info info;
+    if (dladdr((void*)&_layer_dummy_func__, &info)) {
+        dlopen(info.dli_fname, RTLD_NODELETE);
+    }
+}
+#endif
+
+extern "C" {
+    void _layer_dummy_func__() {
```

Comment:
I should be more explicit (and will add comments) but long story short, to work around the Android loader. This will force this layer to NOT get unloaded once it has been loaded once. (There is a lot of global static state in perfetto that explodes if we ever unload the .so)

Reply:
Added a comment